### ISSUE TITLE
Pagination is totally broken when using _sort.

### ISSUE THREAD

**Issue by Natim** [Kinto/kinto#1997]
In that case, the filter are not even filtering anymore and records are returned multiple times.

```py
from kinto_http import Client

client = Client(server_url="https://api-staging.chefclub.tools/v1")

pages = client.get_paginated_records(
    bucket='chefclub', collection='recipes',
    language="fr", _limit=10, _sort="-published_date"
)

records_info = []

try:
    for number, page in enumerate(pages):
        print("Page", number+1)
        for record in page["data"]:
            records_info.append((record["id"], record["title"], record["language"]))
except KeyboardInterrupt:
    pass
print("Unique", len(set([record[0] for record in records_info])))
print("French", len([record for record in records_info if record[2] == "fr"]))
print("Total", len(records_info))
```
Unique 69
French 62
Total 120
```

**Comment by leplatrem** [comment]
I'm surprised, this was supposed to do the job no?

https://github.com/Kinto/kinto/blob/322139b774b356cf024c553affa28548cb3aa694/kinto/core/resource/__init__.py#L1160-L1164

**Comment by Natim** [comment]
I am not sure this is our issue, because even filtering doesn't seems to work. I guess this is bypassed somewhere.

**Comment by Natim** [comment]
What seems crazy to me is that records seems to be by pairs in the same page.

```
Page 16
Meline di bosco
Meline di bosco
Pandoro Fondue!
Pandoro Fondue!
Nutella Xmas Fondue
Nutella Xmas Fondue
Picolé Passatempo
```

**Comment by leplatrem** [comment]
I could not reproduce locally :( 
This is what I did:
https://gist.github.com/leplatrem/af47241da55693e1c75479b2d1e92f48

**Comment by Natim** [comment]
Fixed with #1999
